packages/core/src/processor/util.ts

```diff
@@ -58,3 +58,22 @@ export function selectedLeaf(node: AstNode, offset: number, existingParents: Ast
 	}
 	return null
 }
+
+export function selectedNode(node: AstNode, offset: number): { node: AstNode, parents: AstNode[] } | null
+export function selectedNode(node: AstNode, offset: number, existingParents: AstNode[] = []): { node: AstNode, parents: AstNode[] } | null {
```

Comment:
This is mostly a copy of `selectedLeaf`, but instead of only returning leaves it always returns the deepest node. It's possible these two could be merged.

Reply:
What's the difference between leaf and the deepest node :eyes:

Reply:
Ah I just saw the difference. My old `selectedLeaf` made an assumption that the `children` of a node covers every character of this node, which is fixed by your `selectedNode` implementation. Yeah I think you could just replace the body of `selectedLeaf` with the one of `selectedNode`; that's how it is supposed to work.